src/code/loadfragment2_n64.c

```diff
@@ -0,0 +1,208 @@
+                break;
+
+            case R_MIPS_LO16 << RELOC_TYPE_SHIFT:
+                // Handles relocation for a hi/lo pair, part 2.
+                // Grab the stored LUI (hi) from the `R_MIPS_HI16` section using the `rs` register of the instruction.
+                // The full address is calculated, relocated, and then used to update both the LUI and lo instructions.
+                // If the lo part is negative, add 1 to the LUI value.
+                // Note: The lo instruction is assumed to have a signed immediate.
+
+                luiInstRef = luiRefs[(*relocDataP >> 0x15) & 0x1F];
+                regValP = &luiVals[(*relocDataP >> 0x15) & 0x1F];
+
+                // Check address is valid for relocation
+                if ((((*luiInstRef << 0x10) + (s16)*relocDataP) & 0x0F000000) == 0) {
+                    relocatedAddress = ((*regValP << 0x10) + (s16)*relocDataP) - vramu32 + allocu32;
+                    isLoNeg = (relocatedAddress & 0x8000) ? 1 : 0;
+                    *luiInstRef = (*luiInstRef & 0xFFFF0000) | (((relocatedAddress >> 0x10) & 0xFFFF) + isLoNeg);
+                    *relocDataP = (*relocDataP & 0xFFFF0000) | (relocatedAddress & 0xFFFF);
+                } else if (gOverlayLogSeverity >= 3) {
+                    osSyncPrintf(T("セグメントポインタ16です %08x %08x %08x\n", "Segment pointer 16 %08x %08x %08x\n"),
+                                 ((*luiInstRef << 0x10) + (s16)*relocDataP) - vramu32, *luiInstRef, *relocDataP);
+                }
+                break;
+        }
+    }
+}
+
+size_t Overlay_Load(uintptr_t vromStart, uintptr_t vromEnd, void* vramStart, void* vramEnd, void* allocatedRamAddr) {
+    s32 pad[2];
+    s32 size = vromEnd - vromStart;
+    uintptr_t end;
+    OverlayRelocationSection* ovlRelocs;
+
+    if (gOverlayLogSeverity >= 3) {
+        osSyncPrintf(
+            T("\nダイナミックリンクファンクションのロードを開始します\n", "\nStart loading dynamic link function\n"));
+    }
+
+    size = vromEnd - vromStart;
+    end = (uintptr_t)allocatedRamAddr + size;
+
+    if (gOverlayLogSeverity >= 3) {
+        osSyncPrintf(T("TEXT,DATA,RODATA+relをＤＭＡ転送します(%08x-%08x)\n",
+                       "DMA transfer TEXT, DATA, RODATA+rel (%08x-%08x)\n"),
+                     allocatedRamAddr, end);
+    }
+
+    DmaMgr_RequestSync(allocatedRamAddr, vromStart, size);
+
+    // The overlay file is expected to contain a 32-bit offset from the end of the file to the start of the
+    // relocation section.
+    ovlRelocs = (OverlayRelocationSection*)(end - ((s32*)end)[-1]);
+
+    if (gOverlayLogSeverity >= 3) {
+        osSyncPrintf("TEXT(%08x), DATA(%08x), RODATA(%08x), BSS(%08x)\n", ovlRelocs->textSize, ovlRelocs->dataSize,
+                     ovlRelocs->rodataSize, (s32)ovlRelocs->bssSize);
+    }
+
+    if (gOverlayLogSeverity >= 3) {
+        osSyncPrintf(T("リロケーションします\n", "I will relocate\n"));
+    }
+
+    Overlay_Relocate(allocatedRamAddr, ovlRelocs, vramStart);
+
+    if ((s32)ovlRelocs->bssSize != 0) {
```

Comment:
In every usage of `bssSize` it needs to be an `s32` for ntsc-1.2, but gc version need `u32` which indicates an actual change in the struct. Instead of ifdefing the struct or duplicating the struct with the one change, I decided to just cast it as needed here as it seemed cleaner. Would another method be preferred?

Reply:
works for me but maybe comment this insight somewhere?